packages/superset-ui-legacy-preset-chart-deckgl/src/utils/fitViewport.ts

```diff
@@ -0,0 +1,54 @@
+import { fitBounds } from '@math.gl/web-mercator';
+import computeBoundsFromPoints from './computeBoundsFromPoints';
+
+type Viewport = {
+  longtitude: number;
+  latitude: number;
+  zoom: number;
+  bearing?: number;
+  pitch?: number;
+};
+
+export default function fitViewport(
+  originalViewPort: Viewport,
+  {
+    points,
+    width,
+    height,
+    minExtent,
+    maxZoom,
+    offset,
+    padding = 20,
+  }: {
+    points: [number, number][];
+    width: number;
+    height: number;
+    minExtent?: number;
+    maxZoom?: number;
+    offset?: [number, number];
+    padding?: number;
+  },
+) {
+  try {
+    const { bearing, pitch } = originalViewPort;
+
+    return {
+      ...fitBounds({
+        bounds: computeBoundsFromPoints(points),
+        width,
+        height,
+        minExtent,
+        maxZoom,
+        offset,
+        padding,
+      }),
+      bearing,
+      pitch,
+    };
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.error('Could not fit viewport', error);
+
+    return originalViewPort;
+  }
```

Comment:
Not sure why `try/catch` is needed, but we can at least make it more strict (should be confident that our own functions do not throw errors):

```js
  const { bearing, pitch } = originalViewPort;
  const bonds = computeBoundsFromPoints(points)
  try {
    return {
      ...fitBounds({
        bounds,
        width,
        height,
        minExtent,
        maxZoom,
        offset,
        padding,
      }),
      bearing,
      pitch,
    };
  } catch (error) {
    // eslint-disable-next-line no-console
    console.error('Could not fit viewport', error);
  }
  return originalViewPort;
```

Reply:
sometimes the `fitBounds` function throw error. Have seen that in the past. Good call on reducing `try` scope.